src/geo/transform.js

```diff
@@ -1344,6 +1344,79 @@ class Transform {
             new Point(Number.MAX_VALUE, Number.MAX_VALUE);
     }
 
+    _getGlobeBounds(min: number = 0, max: number = 0): LngLatBounds {
+        const topLeft = new Point(this._edgeInsets.left, this._edgeInsets.top);
+        const topRight = new Point(this.width - this._edgeInsets.right, this._edgeInsets.top);
+        const bottomRight = new Point(this.width - this._edgeInsets.right, this.height - this._edgeInsets.bottom);
+        const bottomLeft = new Point(this._edgeInsets.left, this.height - this._edgeInsets.bottom);
+
+        // Consider far points at the maximum possible elevation
+        // and near points at the minimum to ensure full coverage.
+        const tl = this.pointCoordinate(topLeft, min);
+        const tr = this.pointCoordinate(topRight, min);
+        const br = this.pointCoordinate(bottomRight, max);
+        const bl = this.pointCoordinate(bottomLeft, max);
+
+        const projection = this.projection;
+        const s = Math.pow(2, -this.zoom);
+
+        const x1 = Math.min(tl.x, bl.x);
+        const x2 = Math.max(tr.x, br.x);
+        const y1 = Math.min(tl.y, tr.y);
+        const y2 = Math.max(bl.y, br.y);
+
+        const lng1 = lngFromMercatorX(x1);
+        const lng2 = lngFromMercatorX(x2);
+        const lat1 = latFromMercatorY(y1);
+        const lat2 = latFromMercatorY(y2);
+
+        const p0 = projection.project(lng1, lat1);
+        const p1 = projection.project(lng2, lat1);
+        const p2 = projection.project(lng2, lat2);
+        const p3 = projection.project(lng1, lat2);
+
+        let minX = Math.min(p0.x, p1.x, p2.x, p3.x);
+        let minY = Math.min(p0.y, p1.y, p2.y, p3.y);
+        let maxX = Math.max(p0.x, p1.x, p2.x, p3.x);
+        let maxY = Math.max(p0.y, p1.y, p2.y, p3.y);
+
+        // we pick an error threshold for calculating the bbox that balances between performance and precision
+        const maxErr = s / 16;
+
+        function processSegment(pa, pb, ax, ay, bx, by) {
+            const mx = (ax + bx) / 2;
+            const my = (ay + by) / 2;
+
+            const pm = projection.project(lngFromMercatorX(mx), latFromMercatorY(my));
+            const err = Math.max(0, minX - pm.x, minY - pm.y, pm.x - maxX, pm.y - maxY);
```

Comment:
The `err` is always `0` here since the differences with the midpoint are always negative.

Reply:
The error metric here is a little unclear — can you explain what's measured here and when the recursion is expected to stop in practice, and add a comment perhaps?

Reply:
I'm not sure that it's the best approach here, but this is the same metric we use in `tileTransform`, and the recursion works the same way.

The `err` metric is the maximum distance between the midpoint and each of the currently calculated bounds. 

https://github.com/mapbox/mapbox-gl-js/blob/b7dc6dc935cc2c2abd3c70444f36118bcba7e522/src/geo/projection/tile_transform.js#L59

We expand the bounds on each iteration by the position of the midpoint, and the recursion stops when the error metric fall below the `maxErr` calculated based on the scale.

```js
const s = Math.pow(2, -zoom);
const maxErr = s / 16;
```

https://github.com/mapbox/mapbox-gl-js/blob/b7dc6dc935cc2c2abd3c70444f36118bcba7e522/src/geo/projection/tile_transform.js#L61-L69